# Analisis exploratorio de datos (EDA)

Importamos las librerias necesarias para el desarrollo del analisis exploratorio.

In [33]:
import numpy as np
import pandas as pd
import matplotlib as plt
import seaborn as sns

Importamos el dataset.

In [52]:
hechos = pd.read_excel('homicidios.xlsx')
victimas =pd.read_excel('homicidios.xlsx', sheet_name='VICTIMAS')

Ahora revisamos los datos

In [67]:
hechos.head(3)

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA HH  \
0  2016-0001           1 2016-01-01  2016   1   1  04:00:00  4   
1  2016-0002           1 2016-01-02  2016   1   2  01:15:00  1   
2  2016-0003           1 2016-01-03  2016   1   3  07:00:00  7   

                             LUGAR_DEL_HECHO TIPO_DE_CALLE  ...  Altura  \
0  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...     NaN   
1          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...     NaN   
2                         AV ENTRE RIOS 2034       AVENIDA  ...  2034.0   

                                 Cruce  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                  DE LOS CORRALES AV.   
2                                  NaN   

                               Dirección Normalizada COMUNA  \
0  PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1               PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   
2                                ENTRE RIOS AV. 2034      1   

                                XY (CABA)         pos x         pos y  \
0   Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1   Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2  Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   

    PARTICIPANTES VICTIMA    ACUSADO  
0       MOTO-AUTO    MOTO       AUTO  
1  AUTO-PASAJEROS    AUTO  PASAJEROS  
2       MOTO-AUTO    MOTO       AUTO  

[3 rows x 21 columns]

In [54]:
victimas.head(3)

ID_hecho      FECHA  AAAA  MM  DD        ROL VICTIMA       SEXO EDAD  \
0  2016-0001 2016-01-01  2016   1   1  CONDUCTOR    MOTO  MASCULINO   19   
1  2016-0002 2016-01-02  2016   1   2  CONDUCTOR    AUTO  MASCULINO   70   
2  2016-0003 2016-01-03  2016   1   3  CONDUCTOR    MOTO  MASCULINO   30   

   FECHA_FALLECIMIENTO  
0  2016-01-01 00:00:00  
1  2016-01-02 00:00:00  
2  2016-01-03 00:00:00

Realizamos un .info de los datasets

In [55]:
hechos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 696 entries, 0 to 695
Data columns (total 21 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   ID                     696 non-null    object        
 1   N_VICTIMAS             696 non-null    int64         
 2   FECHA                  696 non-null    datetime64[ns]
 3   AAAA                   696 non-null    int64         
 4   MM                     696 non-null    int64         
 5   DD                     696 non-null    int64         
 6   HORA                   696 non-null    object        
 7   HH                     696 non-null    object        
 8   LUGAR_DEL_HECHO        696 non-null    object        
 9   TIPO_DE_CALLE          696 non-null    object        
 10  Calle                  695 non-null    object        
 11  Altura                 129 non-null    float64       
 12  Cruce                  525 non-null    object        
 13  Direc

Es un dataset con 696 registros y 21 columnas. Tambien se puede ver que hay datos nulos en las columnas de calle, altura, cruce, direcccion normalizada, victima y acusado.

In [56]:
victimas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717 entries, 0 to 716
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   ID_hecho             717 non-null    object        
 1   FECHA                717 non-null    datetime64[ns]
 2   AAAA                 717 non-null    int64         
 3   MM                   717 non-null    int64         
 4   DD                   717 non-null    int64         
 5   ROL                  717 non-null    object        
 6   VICTIMA              717 non-null    object        
 7   SEXO                 717 non-null    object        
 8   EDAD                 717 non-null    object        
 9   FECHA_FALLECIMIENTO  717 non-null    object        
dtypes: datetime64[ns](1), int64(3), object(6)
memory usage: 56.1+ KB


Es un dataset con 717 registros y 10 columnas. Tambien se puede ver que hay datos nulos en las columnas de rol, victima, sexo, edad y fecha_fallecimiento.

Al revisar el pdf de homicidios nos dice que en algunas columnas hay valores con SD los cuales son datos faltantes por lo que se procede a reemplazar estod SD por None.

In [57]:
hechos.replace('SD', None, inplace=True)
victimas.replace('SD', None, inplace=True)

Ademas los 

In [63]:
hechos.replace('.', None, inplace=True)

In [70]:
hechos.head()

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA HH  \
0  2016-0001           1 2016-01-01  2016   1   1  04:00:00  4   
1  2016-0002           1 2016-01-02  2016   1   2  01:15:00  1   
2  2016-0003           1 2016-01-03  2016   1   3  07:00:00  7   
3  2016-0004           1 2016-01-10  2016   1  10  00:00:00  0   
4  2016-0005           1 2016-01-21  2016   1  21  05:20:00  5   

                             LUGAR_DEL_HECHO TIPO_DE_CALLE  ...  Altura  \
0  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...     NaN   
1          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...     NaN   
2                         AV ENTRE RIOS 2034       AVENIDA  ...  2034.0   
3      AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA  ...     NaN   
4  AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA  ...     NaN   

                                 Cruce  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                  DE LOS CORRALES AV.   
2                                  NaN   
3             VILLEGAS, CONRADO, GRAL.   
4              SAENZ PE?A, LUIS, PRES.   

                               Dirección Normalizada COMUNA  \
0  PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1               PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   
2                                ENTRE RIOS AV. 2034      1   
3          LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.      8   
4            SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.      1   

                                 XY (CABA)         pos x         pos y  \
0    Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1    Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2   Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3    Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4  Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   

    PARTICIPANTES VICTIMA    ACUSADO  
0       MOTO-AUTO    MOTO       AUTO  
1  AUTO-PASAJEROS    AUTO  PASAJEROS  
2       MOTO-AUTO    MOTO       AUTO  
3         MOTO-SD    MOTO       None  
4  MOTO-PASAJEROS    MOTO  PASAJEROS  

[5 rows x 21 columns]

Revisamos cuantos datos nulos hay por columnas

In [69]:
hechos.isnull().sum()

ID                         0
N_VICTIMAS                 0
FECHA                      0
AAAA                       0
MM                         0
DD                         0
HORA                       1
HH                         1
LUGAR_DEL_HECHO            1
TIPO_DE_CALLE              0
Calle                      1
Altura                   567
Cruce                    171
Dirección Normalizada      8
COMUNA                     0
XY (CABA)                  0
pos x                     12
pos y                     12
PARTICIPANTES              0
VICTIMA                    9
ACUSADO                   23
dtype: int64

Una vez realizado los cambios podemos ver que ademas de las columnas con datos faltantes anteriormente mencionadas tambien hay datos nulos en las columnas de hora, hh, lugar del hecho y calle.

In [59]:
victimas.isnull().sum()

ID_hecho                0
FECHA                   0
AAAA                    0
MM                      0
DD                      0
ROL                    11
VICTIMA                 9
SEXO                    6
EDAD                   53
FECHA_FALLECIMIENTO    68
dtype: int64

Una vez hecho los cambios no se ven columnas adicionales que presenten datos nulos de las anteriormente identificadas. 

Teniendo en cuenta que hay muy pocos datos se decide trabajar con los datos nulos existentes.

Revisamos si hay datos duplicados.

In [60]:
hechos[hechos.duplicated()]

Empty DataFrame
Columns: [ID, N_VICTIMAS, FECHA, AAAA, MM, DD, HORA, HH, LUGAR_DEL_HECHO, TIPO_DE_CALLE, Calle, Altura, Cruce, Dirección Normalizada, COMUNA, XY (CABA), pos x, pos y, PARTICIPANTES, VICTIMA, ACUSADO]
Index: []

[0 rows x 21 columns]

En el dataframe de hechos no hay datos duplicados.

In [61]:
victimas[victimas.duplicated()]

Empty DataFrame
Columns: [ID_hecho, FECHA, AAAA, MM, DD, ROL, VICTIMA, SEXO, EDAD, FECHA_FALLECIMIENTO]
Index: []

En el dataframe de victimas no hay datos duplicados

In [72]:
hechos.to_excel('hechos.xlsx', index=False)
victimas.to_excel('victimas.xlsx', index=False)

In [51]:
hechos.describe()

N_VICTIMAS         AAAA          MM          DD        Altura  \
count  696.000000   696.000000  696.000000  696.000000    129.000000   
mean     1.030172  2018.188218    6.692529   15.936782   3336.635659   
std      0.179393     1.683754    3.571309    8.639646   3060.641793   
min      1.000000  2016.000000    1.000000    1.000000     30.000000   
25%      1.000000  2017.000000    4.000000    9.000000   1359.000000   
50%      1.000000  2018.000000    7.000000   16.000000   2551.000000   
75%      1.000000  2020.000000   10.000000   23.000000   4500.000000   
max      3.000000  2021.000000   12.000000   31.000000  16080.000000   

           COMUNA  
count  696.000000  
mean     7.425287  
std      4.387050  
min      0.000000  
25%      4.000000  
50%      8.000000  
75%     11.000000  
max     15.000000

In [49]:
hechos.describe(include=object)

ID      HORA   HH                 LUGAR_DEL_HECHO  \
count         696       695  695                             695   
unique        696       374   24                             682   
top     2016-0001  16:00:00    7  AV 27 DE FEBRERO Y AV ESCALADA   
freq            1        10   41                               4   

       TIPO_DE_CALLE           Calle           Cruce  \
count            696             695             525   
unique             4             279             317   
top          AVENIDA  PAZ, GRAL. AV.  PAZ, GRAL. AV.   
freq             429              57              14   

                   Dirección Normalizada    XY (CABA) pos x pos y  \
count                                688          696   696   696   
unique                               635          606   605   605   
top     27 DE FEBRERO AV. y ESCALADA AV.  Point (. .)     .     .   
freq                                   5           14    12    12   

           PARTICIPANTES VICTIMA ACUSADO  
count                696     687     673  
unique                41       9       9  
top     PEATON-PASAJEROS    MOTO    AUTO  
freq                 105     295     204

In [77]:
hechos['ACUSADO'].value_counts()

AUTO           204
PASAJEROS      173
CARGAS         146
OBJETO FIJO     62
MOTO            57
MULTIPLE        17
BICICLETA        7
OTRO             6
TREN             1
Name: ACUSADO, dtype: int64

In [78]:
hechos['VICTIMA'].value_counts()

MOTO           295
PEATON         264
AUTO            83
BICICLETA       29
CARGAS           7
PASAJEROS        5
MOVIL            2
OBJETO FIJO      1
PEATON_MOTO      1
Name: VICTIMA, dtype: int64